tests/python/pytorch/sparse/test_sparse_matrix.py

```diff
@@ -504,6 +504,60 @@ def test_range_select(create_func, shape, dense_dim, select_dim, rang):
     assert torch.allclose(A_select_to_dense, dense_select)
 
 
+@pytest.mark.parametrize(
+    "create_func", [rand_diag, rand_csr, rand_csc, rand_coo]
+)
+@pytest.mark.parametrize("sample_dim", [0, 1])
+@pytest.mark.parametrize("index", [None, (0, 1, 3)])
+@pytest.mark.parametrize("replace", [0, 1])
+@pytest.mark.parametrize("bias", [0, 1])
+def test_sample(create_func, sample_dim, index, replace, bias):
+    ctx = F.ctx()
+    shape = (5, 5)
+    sample_num = 3
+    A = create_func(shape, 6, ctx)
+    A_row = A.row
+    A_col = A.col
+    A_val = torch.abs(A.val)
+    A = from_coo(A_row, A_col, A_val, shape)
+
+    if index is not None:
+        index = torch.tensor(index).to(ctx)
+
+    A_sample = A.sample(sample_dim, sample_num, index, replace, bias)
+    A_dense = sparse_matrix_to_dense(A)
+    A_sample_to_dense = sparse_matrix_to_dense(A_sample)
+
+    if index == None:
+        ans_shape = shape
+        for row in range(shape[0]):
+            ans_ele = list(A_dense[row, :].nonzero().reshape(-1))
+            ret_ele = list(A_sample_to_dense[row, :].nonzero().reshape(-1))
+            for e in ret_ele:
```

Comment:
Could you also test whether the number of elements of each row or column is expected? For example, if a row to sample has `n` nonzero elements and we want to sample `m` among them without replacement, then the sampled size should be `min(n, m)`. You may also need to take care about the case when sampling a row without nonzero elements in a replacement manner.